src/files-and-uploads/FilesAndUploads.jsx

```diff
@@ -0,0 +1,273 @@
+import { useDispatch, useSelector } from 'react-redux';
+import _ from 'lodash';
+import { injectIntl, FormattedMessage, intlShape } from '@edx/frontend-platform/i18n';
+import {
+  DataTable,
+  TextFilter,
+  CheckboxFilter,
+  Dropzone,
+  CardView,
+  useToggle,
+} from '@edx/paragon';
+import Placeholder, { ErrorAlert } from '@edx/frontend-lib-content-components';
+
+import { RequestStatus } from '../data/constants';
+import { useModels } from '../generic/model-store';
+import {
+  addAssetFile,
+  deleteAssetFile,
+  fetchAssets,
+  updateAssetLock,
+} from './data/thunks';
+import messages from './messages';
+
+import FileInput, { fileInput } from './FileInput';
+import FilesAndUploadsProvider from './FilesAndUploadsProvider';
+import {
+  GalleryCard,
+  ListCard,
+  TableActions,
+} from './table-components';
+import ApiStatusToast from './ApiStatusToast';
+
+const FilesAndUploads = ({
+  courseId,
+  // injected
+  intl,
+}) => {
+  const dispatch = useDispatch();
+  const defaultVal = 'card';
+  const columnSizes = { xs: 12, sm: 6, lg: 3 };
+  const [currentView, setCurrentView] = useState(defaultVal);
+  const [isDeleteOpen, setDeleteOpen, setDeleteClose] = useToggle(false);
+  const [isAddOpen, setAddOpen, setAddClose] = useToggle(false);
+  const [selectedRowCount, setSelectedRowCount] = useState(0);
+  const assetIds = useSelector(state => state.assets.assetIds);
+
+  useEffect(() => {
+    dispatch(fetchAssets(courseId));
+  }, [courseId]);
+  const totalCount = useSelector(state => state.assets.totalCount);
+  const addAssetStatus = useSelector(state => state.assets.addingStatus);
+  const deleteAssetStatus = useSelector(state => state.assets.deletingStatus);
+  const saveAssetStatus = useSelector(state => state.assets.savingStatus);
+  const loadingStatus = useSelector(state => state.assets.loadingStatus);
+  const errorMessages = useSelector(state => state.assets.errors);
+  const fileInputControl = fileInput({
+    onAddFile: (file) => dispatch(addAssetFile(courseId, file, totalCount)),
+    setSelectedRowCount,
+    setAddOpen,
+  });
+  const assets = useModels('assets', assetIds);
+
+  const handleDropzoneAsset = ({ fileData, handleError }) => {
+    try {
+      const file = fileData.get('file');
+      dispatch(addAssetFile(courseId, file, totalCount));
+    } catch (error) {
+      handleError(error);
+    }
+  };
+
+  const handleBulkDelete = (selectedFlatRows) => {
+    setSelectedRowCount(selectedFlatRows.length);
+    setDeleteOpen();
+    const assetIdsToDelete = selectedFlatRows.map(row => row.original.id);
+    assetIdsToDelete.forEach(id => dispatch(deleteAssetFile(courseId, id, totalCount)));
+  };
+
+  const handleBulkDownload = (selectedFlatRows) => {
+    selectedFlatRows.forEach(row => {
+      const { externalUrl } = row.original;
+      const link = document.createElement('a');
+      link.target = '_blank';
+      link.download = true;
+      link.href = externalUrl;
+      link.click();
```

Comment:
This opens a bunch of blank tabs which may be disturbing to the user ux wise but i agree its the solution until we can implement zip files.

Looks like the biggest library for this (jszip) hasn't been contributed to in a year, so either is is perfect or dead.

Reply:
yeah that is the library i have been looking into. I plan to work on the zipping after everything is done for the first pass.